pkg/server/datastore/datastore.go

```diff
@@ -0,0 +1,55 @@
+package datastore
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	management "github.com/HewlettPackard/galadriel/pkg/server/api/management"
+	"github.com/labstack/echo/v4"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/proto/spire/common"
+)
+
+// DataStore defines the data storage interface.
+type DataStore interface {
+	CreateMember(echo.Context, *management.SpireServer) (*management.SpireServer, error)
+	CreateMembership(ctx context.Context, membership *management.FederationGroupMembership, memberID uint, bridgeID uint) (*management.FederationGroupMembership, error)
```

Comment:
an example would be:
```suggestion
	CreateMembership(context.Context, membership *management.FederationGroupMembership, MemberID, BridgeID) (*management.FederationGroupMembership, error)
```
and have defined
`type MemberID uint`
`type BridgeID uint`

Reply:
As discussed offline with @thisab, this can be better looked at once we have an updated version of the API specs and definitions. Let's try to have all these types defined in there.